feat: support integration with experiment ios

falconandy opened the pull request:
### Summary

Support integration with experiment ios.
Based on [Amplitude-Kotlin](https://github.com/amplitude/Amplitude-Kotlin/pull/92/files#diff-6d55e2d5810c4d89bd2b78fdec680139ac4b867421ac08ccba98187023c52c6e) and Amplitude-iOS implementation.

Possible issue:
`DEFAULT_INSTANCE` name is `default_instance` in [Amplitude-Swift](https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Constants.swift#L58) and `$default_instance` in [experiment-ios-client](https://github.com/amplitude/experiment-ios-client/blob/main/Sources/Experiment/ExperimentConfig.swift#L82)

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-SDK-Template/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

justin-fiedler reviewed (changes requested):
Thanks @falconandy changes look great. 

> DEFAULT_INSTANCE name is default_instance in [Amplitude-Swift](https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Constants.swift#L58) and $default_instance in [experiment-ios-client](https://github.com/amplitude/experiment-ios-client/blob/main/Sources/Experiment/ExperimentConfig.swift#L82)

Good catch, can you please update Swift to use `$default_instance` to [match the other SDKs](https://amplitude.sourcegraph.com/search?q=context%3Aglobal+%24default_instance&patternType=standard&sm=1&groupBy=repo) (while we're still in Beta).

falconandy commented:
@justin-fiedler 
> Good catch, can you please update Swift to use $default_instance to match the other SDKs (while we're still in Beta).

Updated. Added new migration to move storage data.

justin-fiedler reviewed (approved):
Thanks @falconandy!
